File: java/oop/AccountApp/src/com/techlabs/account/Account.java
package com.techlabs.account;

public class Account {
	private int accNo;
	private String name;
	private double balance=500.0;
	private static int count;
	
	static{
		count=100;
		System.out.println("Inside Static");
	}
	
	public Account(int accNo,String name,double balance){
		this(accNo,name);
		this.balance=balance;
	}
	
	public Account(int accNo,String name){
		this();
		this.accNo=accNo;
		this.name=name;
	}
	
	public Account(){
		Account.count++;
		System.out.println("Object Created.");
	}
	
	public void deposit(double amount){
		this.balance+=amount;
	}
	
	public void withdraw(double amount){
		if(validateWithdraw(amount)){
			this.balance-=amount;
		}
		else{
			System.out.println("Cannot Withdraw Minimum Balance has to be maintained to 500 Rs");
		}
	}
	
	public int getAccountNo(){
		return this.accNo;
	}
	
	public String getName(){
		return this.name;
	}
	
	public double getBalance(){
		return this.balance;
	}
	
	public int getCount(){
		return Account.count;
	}
	
	public static int HeadCount(){
		return Account.count;
	}
	
	public boolean validateWithdraw(double amount){
		if((this.balance-amount)<500)
			return false;
		return true;
	}
	
	@Override
	public String toString(){
		return "Child Class\n Account No: "+this.accNo
				+" - Name: "+this.name
				+" - Balance "+this.balance
				+"\nParent Class\n "+super.toString();
	}
}

File: dotNet/CSharp/AccountAppWithCustomException/AccountAppWithCustomException/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AccountAppWithCustomException
{
    class Account
    {
        private int _accNo;
        private string _name;
        private double _balance = 500.0;
        private static int count;

        static Account()
        {
            count = 100;
            Console.WriteLine("Inside Static");
        }

        public Account(int accNo, string name, double balance) : this(accNo, name)
        {
            this._balance = balance;
        }

        public Account(int accNo, string name) : this()
        {
            this._accNo = accNo;
            this._name = name;
        }

        public Account()
        {
            count++;
            Console.WriteLine("Object Created {0}", count);
        }

        public void Deposit(double amount)
        {
            this._balance += amount;
        }

        public void Withdraw(double amount)
        {
            if (ValidateWithdraw(amount))
            {
                this._balance -= amount;
            }
            else
            {
                throw new InsufficientFundException("Cannot withdraw InsufficientFundException",this);
            }
        }

        private Boolean ValidateWithdraw(double amount)
        {
            if ((this._balance - amount) < 500)
            {
                return false;
            }
            else
            {
                return true;
            }
        }

        public Int32 AccountNo
        {
            get
            {
                return this._accNo;
            }
        }

        public String Name
        {
            get
            {
                return this._name;
            }
        }

        public Double Balance
        {
            get
            {
                return this._balance;
            }
        }

        public override string ToString()
        {
            return "Account No: " + this._accNo
                + "\n Name: " + this._name
                + "\n Balance: " + this._balance;
        }
    }
}
